content/03-reference/01-tools-and-interfaces/01-prisma-schema/07-prisma-schema-reference.mdx

```diff
@@ -862,62 +862,108 @@ N/A
 
 #### Examples
 
+##### Specify a unique attribute on a required `String` field
+
+```prisma
+model User {
+  email String  @unique
+  name  String
+}
+```
+
+##### Specify a unique attribute on an optional `String` field
+
+```prisma
+model User {
+  id    Int     @id @default(autoincrement())
+  email String? @unique
+  name  String
+}
+```
+
+##### Specify a unique attribute on relation scalar field `authorId`
+
+```prisma
+model Post {
+  author    User    @relation(fields: [authorId], references: [id])
+  authorId  Int     @unique
+  title     String
+  published Boolean @default(false)
+}
+
+model User {
+  id    Int     @id @default(autoincrement())
+  email String? @unique
+  name  String
+  Post  Post[]
+}
+```
+
+##### Specify a unique attribute with `cuid()` values as default values
+
+```prisma
+model User {
+  id    String  @unique @default(cuid())
+  name  String
+}
+```
+
 ### <inlinecode>@@unique</inlinecode>
 
 Defines a compound [unique constraint](data-model#defining-a-unique-field) <span class="concept"></span> for the specified fields.
 
 #### Remarks
 
 * Corresponding database type: `UNIQUE`
-* `NULL` values are considered to distinct (multiple `NULL` values in a column are allowed)
-* Can be defined on any scalar field
-* **Cannot** be defined on a relation field
-* A field annotated with `@@unique` can be optional or required 
-* A field annotated with `@@unique` _must_ be required if it represents the only unique constraint on a model without an `@id` / `@@id`
-* A model can have any number of unique constraints
+* A `@@unique` block is required if it represents the only unique constraint on a model without an `@id` / `@@id`
+* All fields that make up the unique constraint **must** be mandatory fields. The following model is **not** valid because `id` could be `null`:
+
+  ```prisma
+  model User {
+    firstname Int
+    lastname Int
+    id Int?
+    @@unique([firstname,lastname,id])
+  }
+  ```
```

Comment:
Semi offtopic:
What is our general stance on "wrong" examples? Do we just put them in and note in the text that they are _not_ what people are supposed to do?

Reply:
There's no general stance - I tend to add a comment next to specific line and say 'this will NOT work' - perhaps I will do that here. But that's interesting - I will chat to Matej about whether we should do something design-related:

![image](https://user-images.githubusercontent.com/849648/95206501-1076e800-07e7-11eb-8dd1-7a9d23792051.png)

Reply:
I've added this to Design's inbox.

Reply:
Coming in from Design inbox -> Is generally the entire example 'wrong'? Like is it specifically a counter-example? Or is there usually a line that's worth highlighting as the main culprit? Or both?

Reply:
@mhwelander @nikolasburk